Tests/BuildTests/BuildPlanTests.swift

```diff
@@ -2107,6 +2108,54 @@ final class BuildPlanTests: XCTestCase {
         XCTAssertMatch(exe, [.anySequence, "-L", "/path/to/foo", "-L/path/to/foo", "-Xlinker", "-rpath=foo", "-Xlinker", "-rpath", "-Xlinker", "foo", "-L", "/fake/path/lib"])
     }
 
+    func testUserToolchainCompileFlags() throws {
+        let fs = InMemoryFileSystem(emptyFiles:
+            "/Pkg/Sources/exe/main.swift",
+            "/Pkg/Sources/lib/lib.c",
+            "/Pkg/Sources/lib/include/lib.h"
+        )
+
+        let diagnostics = DiagnosticsEngine()
+        let graph = try loadPackageGraph(fs: fs, diagnostics: diagnostics,
+            manifests: [
+                Manifest.createV4Manifest(
+                    name: "Pkg",
+                    path: "/Pkg",
+                    packageKind: .root,
+                    packageLocation: "/Pkg",
+                    targets: [
+                        TargetDescription(name: "exe", dependencies: ["lib"]),
+                        TargetDescription(name: "lib", dependencies: []),
+                    ]),
+            ]
+        )
+        XCTAssertNoDiagnostics(diagnostics)
+
+        let userDestination = Destination(sdk: AbsolutePath("/fake/sdk"),
+            binDir: Resources.default.toolchain.destination.binDir,
```

Comment:
Updated this line to use this toolchain bindir rather than the fake path I had before, which should get it working on the Linux smoke test CI again, as it works locally for me now.